src/main/java/Reacher.java

```diff
@@ -0,0 +1,91 @@
+                        if (num > memory.size() || num < 1) {
+                            throw new ReacherException("No such task number");
+                        }
+                        Task task = memory.get(num - 1);
+                        System.out.println("Mark Done or Undone or Delete?");
+                        String change = scanner.nextLine();
+                        if (change.equalsIgnoreCase("done")) {
+                            task.markDone();
+                            System.out.println("Task " + num + " marked done");
+                        } else if (change.equalsIgnoreCase("undone")) {
+                            task.markNotDone();
+                            System.out.println("Task " + num + " marked Undone");
+                        } else if (change.equalsIgnoreCase("delete")) {
+                            memory.remove(num);
+                            System.out.println("Task " + num + " deleted");
+                        }else {
+                            throw new ReacherException("u did not write done, undone or delete.");
+                        }
+                    } catch (Exception e) {
+                        System.out.println(e.getMessage());
+                    }
+                } else {
+                    System.out.println("What type of task is this?(Deadline, Event, Todo)");
+                    try {
+                        String type = scanner.nextLine();
+                        if (type.equalsIgnoreCase("todo")) {
+                            Todos t = new Todos(input);
+                            memory.add(t);
+                            System.out.println("I've added " + t.toString());
+                        } else if (type.equalsIgnoreCase("deadline")) {
+                            System.out.println("When is the deadline?");
+                            String deadline = scanner.nextLine();
+                            Deadline t = new Deadline(input, deadline);
+                            memory.add(t);
+                            System.out.println("I've added " + t.toString());
+                        } else if (type.equalsIgnoreCase("event")) {
+                            System.out.println("When is the start?");
+                            String start = scanner.nextLine();
+                            System.out.println("When is the end?");
+                            String end = scanner.nextLine();
+                            Events t = new Events(input, start, end);
+                            memory.add(t);
+                            System.out.println("I've added " + t.toString());
+                        } else {
+                            throw new ReacherException("That is not a type of task.");
+                        }
+                    } catch (ReacherException e) {
+                        System.out.println(e.getMessage());
+                    }
+                }
+            } catch (Exception e) {
```

Comment:
I like that you added a catch Exception to catch all the exceptions that might have gone past ReacherException!